lib/Dialect/TritonGPU/IR/Dialect.cpp

```diff
@@ -1455,14 +1478,18 @@ SmallVector<unsigned> AMDMfmaEncodingAttr::getSizePerThread() const {
     llvm_unreachable("Unexpected mfma non-k dim");
 
   if (getIsTransposed()) {
-    return {cols, rows};
+    res[rank - 1] = rows;
+    res[rank - 2] = cols;
   } else {
-    return {rows, cols};
+    res[rank - 1] = cols;
+    res[rank - 2] = rows;
   }
+  return res;
 }
 
 SmallVector<int64_t>
 AMDMfmaEncodingAttr::getMFMAInstrShapeForOperands(int kWidth, int opIdx) const {
+  // TODO(lixun): should we extend MFMAInstrShape to 3D?
```

Comment:
I didn't have to do it for `mma`, can we remove the todo ?

Reply:
Sure